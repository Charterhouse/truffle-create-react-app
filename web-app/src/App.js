import React from 'react'
import { BrowserRouter as Router, Route } from 'react-router-dom'
import { Home } from 'pages/home'
import { Accounts } from 'pages/accounts'
import { DApp } from 'pages/dapp'
import { Web3Loader } from 'components/web3'

const renderComponent = (Component, routeProps, web3Props) => (
  <Component {...web3Props}
    {...routeProps} />
)

const web3IsReady = ({web3, accounts, contract}) => {
  return (web3 && accounts && contract)
}

export default () =>
  <Router>
    <Web3Loader render={web3Props => {
      if (web3IsReady(web3Props)) {
        return (
          <div>
            <Route exact path='/' component={Home} />
            <Route path='/dapp' render={routeProps => renderComponent(DApp, routeProps, web3Props)} />
            <Route path='/accounts' render={routeProps => renderComponent(Accounts, routeProps, web3Props)} />
          </div>
        )
      } else {
        return <p>Loading web3, accounts, and contract.</p>
      }
    }} />
  </Router>
