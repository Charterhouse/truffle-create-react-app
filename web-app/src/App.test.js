import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import App from './App'

jest.mock('components/web3/Web3Loader')

it('renders without crashing', async () => {
  const div = document.createElement('div')
  await ReactDOM.render(<App />, div)
})

it('renders correctly', async () => {
  const app = await renderer
    .create(<App />)
  expect(app.toJSON()).toMatchSnapshot()
})
