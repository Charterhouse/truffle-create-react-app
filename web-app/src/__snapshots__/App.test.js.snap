// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    style={
      Object {
        "padding": "20px",
      }
    }
  >
    <h1>
      Home
    </h1>
    <p>
      Note that Web3 is already loaded.
    </p>
    <div
      style={
        Object {
          "marginTop": "20px",
        }
      }
    >
      <a
        href="/dapp"
        onClick={[Function]}
      >
        My DApp
      </a>
      <a
        href="/accounts"
        onClick={[Function]}
        style={
          Object {
            "marginLeft": "20px",
          }
        }
      >
        My Accounts
      </a>
    </div>
  </div>
</div>
`;
